docs/_snippets/architecture/collection.md

```diff
@@ -0,0 +1,24 @@
+```dart
+class YourCollection extends FirefuelCollection<YourModel> {
+  YourCollection() : super('yourCollectionName');
+
+  @override
+  YourModel? fromFirestore(
+      DocumentSnapshot<Map<String, dynamic>> snapshot,
+      SnapshotOptions? options,
+  ) {
+    final data = snapshot.data();
+
+    if(data == null) return null;
+
+    // If you want your model to contain the DocumentId, make your `fromJson` 
+    // function accept the `snapshot.id` and assign it as a property on your model!
+    return YourModel.fromJson(snapshot.data()!, snapshot.id); 
```

Comment:
I think I would want to go down the route of creating a new object rather than mutating the existing one, but I see what you're saying. I'll add another section for this as an alternative.

```dart
fromFirestore(json, id) {
  final jsonWithId = {...json, 'id': id};

   return fromJson(jsonWithId);
}

```

Reply:
We'll probably want to create an example of how to use this with Freezed anyway